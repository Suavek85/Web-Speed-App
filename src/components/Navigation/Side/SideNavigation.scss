@import '../../../scss/styles';

$block: nav-wrapper;
$blockBackground: elem($block, backgrounds);
$blockContent: elem($block, content);

.#{$block} {
  position: absolute;
  top: 0;
  left: -300px;
  z-index: 100;
  display: block;
  width: 0;
  height: 100vh;
  background-color: var(--color-innermost-background);
}

.#{$blockBackground} {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 10;
  display: none;
  width: 100%;
  height: 100%;
  background-color: $color-black;
  opacity: 0;
}

.#{$blockContent} {
  display: flex;
  align-items: center;
  justify-content: center;
}

:export {
  block: $block;
  blockBackground: $blockBackground;
  blockContent: $blockContent;
}
