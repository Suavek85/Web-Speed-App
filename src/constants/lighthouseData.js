export const lighthouseData = [
  'first-contentful-paint',
  'largest-contentful-paint',
  'speed-index',
  'interactive', 
  'total-blocking-time',
  'cumulative-layout-shift'
]

//largest-contentful-paint

//Largest Contentful Paint element
//Largest Contentful Paint marks the time at which the largest text or image is painted.


//first-contentful-paint

//First Contentful Paint marks the time at which the first text or image is painted

//total-blocking-time

//"Sum of all time periods between FCP and Time to Interactive, when task length exceeded 50ms, expressed in milliseconds


//cumulative-layout-shift
//Cumulative Layout Shift measures the movement of visible elements within the viewport.


        