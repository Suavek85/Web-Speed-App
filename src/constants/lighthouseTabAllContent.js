export const TITLE = 'Title:'
export const DESCRIPTION = 'Description:'
export const SCORE = 'Score:'
export const DISPLAY_VALUE = 'Display Value:'