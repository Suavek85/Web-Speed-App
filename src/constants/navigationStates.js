const states = [
  'EXPAND',
  'COLLAPSE',
  'NOANIMATION'
]
      
export const navigationStates = states.reduce((acc, next) => {
  acc[next] = next
  return acc
}, {})