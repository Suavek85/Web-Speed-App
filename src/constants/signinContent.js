export const CLOSE = 'Close'
export const ALREADY_ACCOUNT = 'I already have an account'
export const NO_ACCOUNT = `I don't have an account`
export const SIGN_IN_PROMPT = 'Sign in with your email and password'
export const SIGN_UP_PROMPT = 'Sign up with your email and password'
export const USERNAME_LABEL = 'Username'
export const EMAIL_LABEL = 'Email'
export const PASSWORD_LABEL = 'Password'
export const CONFIRM_PASSWORD_LABEL = 'Confirm password'
export const SIGNIN_TAB = 'Sign in'
export const SIGNUP_TAB = 'New account'
export const SUBMIT = 'Submit'