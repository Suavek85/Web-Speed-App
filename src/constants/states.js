const states = [
  'LOADING',
  'INACTIVE',
  'ERROR',
  'SUCCESS'
]
    
export const appStates = states.reduce((acc, next) => {
  acc[next] = next
  return acc
}, {})