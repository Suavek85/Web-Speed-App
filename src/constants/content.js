import { appStates } from './states'

//APP HEADER
export const APPTITLE = 'Web Speed App'
export const SETTINGS = 'Settings'
export const SIGN_IN = 'Sign in'


//BUTTONS
export const btnsContent = {...appStates}
btnsContent.LOADING = 'Loading...' 
btnsContent.INACTIVE = 'Get report' 
btnsContent.ERROR = 'Oops, an error' 
btnsContent.SUCCESS = 'Get new report'

//LIGHTHOUSE

export const LIGHTHOUSE_TITLE = 'Lighthouse Score'
export const PERFORMANCE = 'Performance'

//FIRST CONTENTFUL PAIN
export const FIRST_PAINT_HEADER = 'First Contentful Paint'
export const FIRST_PAINT_CONTENT = 'First Contentful Paint (FCP) measures the time from navigation to the time when the browser renders the first bit of content from the DOM. This is an important milestone for users because it provides feedback that the page is actually loading.'
export const RESULT = 'Result: '
export const INPUT_HINT = 'Enter your website url in this format https://www.bbc.co.uk/'

//FIRST INPUT DELAY
export const FIRST_INPUT_DELAY_HEADER = 'First Input Delay'
export const FIRST_INPUT_DELAY_CONTENT = 'First Input Delay (FID) measures the time from when a user first interacts with your site (i.e. when they click a link, tap on a button, or use a custom, JavaScript-powered control) to the time when the browser is actually able to respond to that interaction.'


//LIGHTHOUSE TABLE

//FIRST PAINT - LIGHTHOUSE
const FIRST_PAINT_TITLE = 'First Contentful Paint'
const FIRST_PAINT_DESC = 'First Contentful Paint marks the time at which the first text or image is painted.'

//LARGEST PAINT - LIGHTHOUSE
const LARGEST_PAINT_TITLE = 'Largest Contentful Paint'
const LARGEST_PAINT_DESC = 'Largest Contentful Paint marks the time at which the largest text or image is painted.'

//SPEED INDEX - LIGHTHOUSE
const SPEED_INDEX_TITLE = 'Speed Index.'
const SPEED_INDEX_DESC = 'Speed Index shows how quickly the contents of a page are visibly populated.'

//TIME TO INTERACTIVE - LIGHTHOUSE
const INTERACTIVE_TITLE = 'Time to Interactive.'
const INTERACTIVE = 'Time to interactive is the amount of time it takes for the page to become fully interactive.'

//TOTAL BLOCKING TIME - LIGHTHOUSE
const BLOCKING_TIME_TITLE = 'Total Blocking Time'
const BLOCKING_TIME_DESC = 'Sum of all time periods between FCP and Time to Interactive, when task length exceeded 50ms, expressed in milliseconds'

//CUMULATIVE LAYOUT SHIFT
const LAYOUT_SHIFT_TITLE = 'Cumulative Layout Shift.'
const LAYOUT_SHIFT_DESC = 'Cumulative Layout Shift measures the movement of visible elements within the viewport.'

//TABS HEADERS
export const tabsHeader = [
  { content: 'First Paint' },
  { content: 'Largest Paint' },
  { content: 'Speed Index' },
  { content: 'Interactive' },
  { content: 'Blocking Time' },
  { content: 'Layout Shift' },
]

//TABS TITLES
export const tabsTitles = [
  FIRST_PAINT_TITLE, 
  LARGEST_PAINT_TITLE, 
  SPEED_INDEX_TITLE, 
  INTERACTIVE_TITLE,
  BLOCKING_TIME_TITLE,
  LAYOUT_SHIFT_TITLE
]

//TABS DESCRIPTIONS
export const tabsDescriptions = [
  FIRST_PAINT_DESC, 
  LARGEST_PAINT_DESC, 
  SPEED_INDEX_DESC, 
  INTERACTIVE,
  BLOCKING_TIME_DESC,
  LAYOUT_SHIFT_DESC
]
