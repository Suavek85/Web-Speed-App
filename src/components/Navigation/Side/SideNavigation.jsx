import React from 'react'
import { useSpring, animated } from 'react-spring'
import { useSelector, useDispatch } from "react-redux"
import { useMediaQuery } from 'react-responsive'
import { desktopWidth } from'../../../scss/mediaqueries'
import Button from '../../../utils/Button/Button'
import ToggleSwitch from '../../ToggleSwitch/ToggleSwitch'
import MenuLink from '../../../utils/MenuLink/MenuLink'
import { SIGN_IN } from '../../../constants/content'
import styles from './SideNavigation.scss'
import { 
  collapseNavConfig, 
  expandNavConfig, 
  collapseBackgroundConfig, 
  expandBackgroundConfig } from './animatedStyles'
import { navigationStates } from '../../../constants/navigationStates'


function SideNavigation() {

  const getToggleState = useSelector(state => state.toggleReducer.toggleNavigation)

  //using useSpring instead of useTransition becouse of react-spring beta version bug
  const collapseNav = useSpring(collapseNavConfig)
  const expandNav = useSpring(expandNavConfig)
  const collapseBackground = useSpring(collapseBackgroundConfig)
  const expandBackground = useSpring(expandBackgroundConfig)


  const dispatch = useDispatch()
  const handleCollapseMenu = () => dispatch({ type: navigationStates.COLLAPSE })

  const getAnimation = (getToggleState, expandAnimation, collapseAnimation) => {
    if (getToggleState === navigationStates.EXPAND) return expandAnimation
    if (getToggleState === navigationStates.COLLAPSE ) return collapseAnimation
    return
  }

  const isDesktop = useMediaQuery(desktopWidth)

  return (
    <>
      <animated.div 
        style={ getAnimation(getToggleState, expandBackground, collapseBackground) } 
        className={ styles.blockBackground } >
      </animated.div>
      <animated.div 
        style={ getAnimation(getToggleState, expandNav, collapseNav) } 
        className={ styles.block } 
      >
        <div className={ styles.blockContent } style={{transform: 'translate(15px, 15px)',}}>
          <Button 
            btnStyle='oval' 
            handleOnClick={ handleCollapseMenu }
            isDesktopCloseContent
          />
          {!isDesktop && (
            <>
              <MenuLink linkPath="/signin" content={ SIGN_IN } />
              <ToggleSwitch />
            </>
          )}
        </div>
      </animated.div>
    </>
  )}
  
export default React.memo(SideNavigation)
