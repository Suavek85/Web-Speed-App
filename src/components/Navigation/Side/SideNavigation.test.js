import * as React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import SideNavigation from './SideNavigation.jsx'
import { navigationStates } from '../../../constants/navigationStates'

describe('SideNavigation snapshot', () => {

  it('should match the snapshot', () => {
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    const mockDispatch = jest.fn();
    const spyUseDispatch = jest.spyOn(redux, 'useDispatch')
    spyUseSelector.mockReturnValue(navigationStates.EXPAND)
    spyUseDispatch.mockReturnValue(mockDispatch)
    const wrapper = shallow(<SideNavigation />)
    expect(wrapper).toMatchSnapshot()
  })

})
