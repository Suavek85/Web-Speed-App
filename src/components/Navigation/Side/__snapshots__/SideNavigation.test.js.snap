// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideNavigation snapshot should match the snapshot 1`] = `
<Fragment>
  <Animated(div)
    style={
      Object {
        "display": SpringValue {
          "_children": Set {},
          "_defaultProps": Object {},
          "_lastCallId": 0,
          "_lastToId": 0,
          "_phase": "CREATED",
          "_priority": 0,
          "_state": Object {
            "pauseQueue": Set {},
            "resumeQueue": Set {},
          },
          "animation": Animation {
            "changed": false,
            "config": AnimationConfig {
              "bounce": undefined,
              "clamp": false,
              "damping": 1,
              "decay": undefined,
              "duration": undefined,
              "easing": [Function],
              "frequency": undefined,
              "friction": 26,
              "mass": 1,
              "precision": undefined,
              "progress": undefined,
              "restVelocity": undefined,
              "round": undefined,
              "tension": 170,
              "velocity": 0,
            },
            "from": undefined,
            "fromValues": Array [],
            "immediate": false,
            "onChange": undefined,
            "onRest": Array [],
            "onStart": undefined,
            "to": undefined,
            "toValues": null,
            "values": Array [],
          },
          "id": 8,
          "key": "display",
          "queue": undefined,
        },
        "opacity": SpringValue {
          "_children": Set {},
          "_defaultProps": Object {},
          "_lastCallId": 0,
          "_lastToId": 0,
          "_phase": "CREATED",
          "_priority": 0,
          "_state": Object {
            "pauseQueue": Set {},
            "resumeQueue": Set {},
          },
          "animation": Animation {
            "changed": false,
            "config": AnimationConfig {
              "bounce": undefined,
              "clamp": false,
              "damping": 1,
              "decay": undefined,
              "duration": undefined,
              "easing": [Function],
              "frequency": undefined,
              "friction": 26,
              "mass": 1,
              "precision": undefined,
              "progress": undefined,
              "restVelocity": undefined,
              "round": undefined,
              "tension": 170,
              "velocity": 0,
            },
            "from": undefined,
            "fromValues": Array [],
            "immediate": false,
            "onChange": undefined,
            "onRest": Array [],
            "onStart": undefined,
            "to": undefined,
            "toValues": null,
            "values": Array [],
          },
          "id": 7,
          "key": "opacity",
          "queue": undefined,
        },
      }
    }
  />
  <Animated(div)
    style={
      Object {
        "left": SpringValue {
          "_children": Set {},
          "_defaultProps": Object {},
          "_lastCallId": 0,
          "_lastToId": 0,
          "_phase": "CREATED",
          "_priority": 0,
          "_state": Object {
            "pauseQueue": Set {},
            "resumeQueue": Set {},
          },
          "animation": Animation {
            "changed": false,
            "config": AnimationConfig {
              "bounce": undefined,
              "clamp": false,
              "damping": 1,
              "decay": undefined,
              "duration": undefined,
              "easing": [Function],
              "frequency": undefined,
              "friction": 26,
              "mass": 1,
              "precision": undefined,
              "progress": undefined,
              "restVelocity": undefined,
              "round": undefined,
              "tension": 170,
              "velocity": 0,
            },
            "from": undefined,
            "fromValues": Array [],
            "immediate": false,
            "onChange": undefined,
            "onRest": Array [],
            "onStart": undefined,
            "to": undefined,
            "toValues": null,
            "values": Array [],
          },
          "id": 3,
          "key": "left",
          "queue": undefined,
        },
        "width": SpringValue {
          "_children": Set {},
          "_defaultProps": Object {},
          "_lastCallId": 0,
          "_lastToId": 0,
          "_phase": "CREATED",
          "_priority": 0,
          "_state": Object {
            "pauseQueue": Set {},
            "resumeQueue": Set {},
          },
          "animation": Animation {
            "changed": false,
            "config": AnimationConfig {
              "bounce": undefined,
              "clamp": false,
              "damping": 1,
              "decay": undefined,
              "duration": undefined,
              "easing": [Function],
              "frequency": undefined,
              "friction": 26,
              "mass": 1,
              "precision": undefined,
              "progress": undefined,
              "restVelocity": undefined,
              "round": undefined,
              "tension": 170,
              "velocity": 0,
            },
            "from": undefined,
            "fromValues": Array [],
            "immediate": false,
            "onChange": undefined,
            "onRest": Array [],
            "onStart": undefined,
            "to": undefined,
            "toValues": null,
            "values": Array [],
          },
          "id": 4,
          "key": "width",
          "queue": undefined,
        },
      }
    }
  >
    <div
      style={
        Object {
          "transform": "translate(15px, 15px)",
        }
      }
    >
      <Button
        btnStyle="oval"
        handleOnClick={[Function]}
        isDesktopCloseContent={true}
      />
      <MenuLink
        content="Sign in"
        linkPath="/signin"
      />
      <ToggleSwitch />
    </div>
  </Animated(div)>
</Fragment>
`;
